src/main/java/org/kaazing/netx/ws/internal/WsURLConnectionImpl.java

```diff
@@ -775,29 +657,19 @@ private void disconnect() {
         }
     }
 
-    private void readControlFramePayload(Frame controlFrame, long payloadLength) throws IOException {
-        InputStream in = connection.getInputStream();
-        int offset = controlFrame.getDataOffset();
-        int length = (int) payloadLength;
-
-        while (length > 0) {
-            int bytesRead = in.read(controlFrame.buffer().array(), offset, length);
-
-            if (bytesRead == -1) {
-                doFail(WS_PROTOCOL_ERROR, MSG_END_OF_STREAM);
-            }
-
-            offset += bytesRead;
-            length -= bytesRead;
-        }
-    }
-
     private byte[] randomBytes(int size) {
         byte[] bytes = new byte[size];
         random.nextBytes(bytes);
         return bytes;
     }
 
+    private void sendClose(int code, byte[] reason, int offset, int length) throws IOException {
+        getOutputStream().writeClose(code, reason, offset, length);
+        disconnect();
+        inputState = CLOSED;
```

Comment:
Is input state truly closed immediately when sending closed, or does that only occur when close frame is received (or connection dropped)?